src/register.jl

```diff
@@ -109,6 +109,172 @@ struct RegBranch
+            end
+        catch ex
+            if isa(ex, ArgumentError)
+                err = "Error in `[compat]`: $(ex.msg)"
+                @debug(err)
+                throw(err)
+            else
+                rethrow(ex)
+            end
+        end
+    end
+
+    @debug("update package data: compat file")
+    compat_file = joinpath(package_path, "Compat.toml")
+    if isfile(compat_file)
+        compat_data = Pkg.Compress.load(compat_file)
+    else
+        compat_data = Dict()
+    end
+    compat_data[pkg.version] = Dict{String,Any}(n=>Pkg.Types.semver_spec(v) for (n,v) in pkg.compat)
+    Pkg.Compress.save(compat_file, compat_data)
+
+    return
+end
+
+function get_package_path(registry_path, pkg::Pkg.Types.Project)
+    @debug("find package in registry")
+    registry_file = joinpath(registry_path, "Registry.toml")
+    registry_data = TOML.parsefile(registry_file)
+
+    uuid = string(pkg.uuid)
+    if haskey(registry_data["packages"], uuid)
+        package_data = registry_data["packages"][uuid]
+        if (package_data["name"] != pkg.name)
+            err = "Changing package names not supported yet"
+            @debug(err)
+            throw(err)
+        end
+        package_path = joinpath(registry_path, package_data["path"])
+    else
+        @debug("Package with UUID: $uuid not found in registry, checking if UUID was changed")
+        for (k, v) in registry_data["packages"]
+            if v["name"] == pkg.name
+                err = "Changing UUIDs is not allowed"
+                @debug(err)
+                throw(err)
+            end
+        end
+
+        @debug("Creating directory for new package $(pkg.name)")
+        first_letter = uppercase(pkg.name[1])
+        package_path = joinpath(registry_path, "$first_letter", pkg.name)
+        mkpath(package_path)
+    end
+
+    return package_path
+end
+
+function commit_registry(pkg::Pkg.Types.Project, package_path, package_repo, tree_hash, git)
+    @debug("commit changes")
+    message = """
+    New version: $(pkg.name) v$(pkg.version)
+
+    UUID: $(pkg.uuid)
+    Repo: $(package_repo)
+    Tree: $(string(tree_hash))
+    """
+    run(`$git add --all`)
+    run(`$git commit -qm $message`)
+end
+
+function get_tree_hash(package_path, gitconfig)
+    # repo = LibGit2.GitRepo(package_repo)
+    # commit = LibGit2.GitObject(repo, "HEAD")
+    # tree = LibGit2.peel(commit)
+    # tree_hash = string(LibGit2.GitHash(git_tree))
+
+    git = gitcmd(package_path, gitconfig)
+    return read(`$git log --pretty=format:%T -1`, String)
+end
+
+function get_remote_repo(package_path, gitconfig)
+    git = gitcmd(package_path, gitconfig)
+    remote_name = split(chomp(read(`$git remote`, String)), '\n')
+    length(remote_name) > 1 && throw("Repository has multiple remotes.")
+    # TODO: Handle this case?
+    remote_name[1] == "" && throw("Repository does not have a remote.")
+    return read(`$git remote get-url $(remote_name[1])`, String)
```

Comment:
This should `chomp` the ending newline? (`readchomp` can be used directly if the `String` argument is removed.)